dwave/gate/qir/__init__.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""Quantum Intermediate Representation (QIR) module.
+
+Contains the QIR compiler, loader and related files and functions.
+"""
+
+from typing import Tuple
+
+try:
+    import pyqir  # noqa: F401
+except ImportError:  # pragma: no cover
+    pyqir_installed = False
+else:
+    # pyqir>=0.8 required
+    # after deprecating py3.7 use importlib.metadata.version instead'
+    from pkg_resources import get_distribution
+    version = lambda v: get_distribution(v).version  # noqa: E731
+
+    def parse(version: str) -> Tuple[str, str, str]:
+        "Parse a MAJOR.MINOR.PATCH version string into a tuple."
+        return tuple(version.split("."))[:3]
+
+    pyqir_installed = parse(version("pyqir")) >= parse("0.8.0")
```

Comment:
Just a safety check to make sure that the version is at least 0.8. You could otherwise have an incompatible version of PyQIR installed on the system and get weird and annoying errors.

Reply:
No, what I meant, if you define `pyqir>=0.8.0` dependency in package's setup.py, and that is not satisfied during runtime, you'll get `pkg_resources.ContextualVersionConflict` raised on `dwave.gate` import.

Reply:
Yes, but we don't have `pyqir` as a dependency in setup.py due to [aarch64 not being supported](https://github.com/qir-alliance/pyqir/issues/215#issuecomment-1542456786), thus checking whether it's installed or not.

Reply:
Ah, I wasn't aware of that. And given the fact you're [failing hard](https://github.com/dwavesystems/dwave-gate/blob/926051dda7887b95d10eb761698cc12eedf9b728/dwave/gate/qir/__init__.py#L39) on import if `pyqir` is missing, it seemed easier to just add it to the install dependencies.

So you want to make sure dwave-gate can be installed (e.g. as part of Ocean), but not necessarily imported?

Reply:
dwave-gate should be installable and usable without pyqir. It should only fail if trying to import anything from the QIR module, so the rest should work fine even if `pyqir` isn't installed.

Reply:
Ok, that sounds good. In that case, this check makes sense.